File: backend/server/src/main/java/server/utils/RegExpPatterns.java
package server.utils;

import java.util.regex.Pattern;

// TODO: 17.11.2022 include const instead of regexp in project
public class RegExpPatterns {

    static Pattern PHONE_OR_EMAIL_PATTERN = Pattern.compile("^\\+?[0-9]*$|^[A-Z]*$");

    public static String PHONE_OR_EMAIL = PHONE_OR_EMAIL_PATTERN.pattern();
}

File: frontend/src/app/validators/RegExpPatterns.ts
export const regExpPatterns = {
  phonePattern: RegExp('^(\\+7|7|8)?\\d{10}$'),
  emailPattern: RegExp('^(?=.{1,64}@)[A-Za-z0-9_-]+(\\.[A-Za-z0-9_-]+)*@[^-][A-Za-z0-9-]+(\\.[A-Za-z0-9-]+)*(\\.[A-Za-z]{2,})$'),
  passwordPattern: RegExp('^.*(?=.*\\d)(?=.*[a-zа-яё])(?=.*[A-ZА-ЯЁ]).*$')
}
